src/ansible_navigator/actions/exec.py

```diff
@@ -0,0 +1,102 @@
+"""Run the :exec subcommand."""
+import os
+import shlex
+
+from typing import List
+from typing import Optional
+from typing import Tuple
+from typing import Union
+
+from . import _actions as actions
+from ..app import App
+from ..configuration_subsystem import ApplicationConfiguration
+from ..runner import Command
+
+
+@actions.register
+class Action(App):
+    """Run the :exec subcommand."""
+
+    # pylint: disable=too-few-public-methods
+
+    KEGEX = r"^e(?:xec)?$"
+
+    def __init__(self, args: ApplicationConfiguration):
+        """Initialize the action.
+
+        :param args: The current application configuration.
+        """
+        super().__init__(args=args, logger_name=__name__, name="exec")
+
+    @staticmethod
+    def _generate_command(exec_command: str, exec_shell: bool) -> Tuple[str, Optional[List[str]]]:
+        """Generate the command and args.
+
+        :param exec_command: The command to run
+        :param exec_shell: Should the command be wrapped in a shell
+        :returns: The command and any pass through arguments
+        """
+        pass_through_args = None
+        if exec_shell and exec_command:
+            command = "/bin/bash"
+            pass_through_args = ["-c", exec_command]
+        else:
+            parts = shlex.split(exec_command)
+            command = parts[0]
+            if len(parts) > 1:
+                pass_through_args = parts[1:]
+        return (command, pass_through_args)
+
+    def run_stdout(self) -> Union[None, int]:
+        """Run in mode stdout.
+
+        :returns: The return code or None
+        """
+        self._logger.debug("exec requested in stdout mode")
+        response = self._run_runner()
+        if response:
+            _, _, ret_code = response
+            return ret_code
+        return None
+
+    def _run_runner(self) -> Optional[Tuple]:
+        # pylint: disable=too-many-branches
+        # pylint: disable=too-many-statements
+        """Spin up runner.
+
+        :return: The stdout, stderr and return code from runner
+        """
+        if isinstance(self._args.set_environment_variable, dict):
+            set_envvars = {**self._args.set_environment_variable}
```

Comment:
I think there's no point in this since it's already a dict, right?
```suggestion
            set_envvars = self._args.set_environment_variable
```

Reply:
this is what is great about someone else seeing, that was a cut and paste from another action without looking at it twice. yeah, because we manipulate them a copy is cool